### PR TITLE
add log spinner when setting up the cluster

### PR DESCRIPTION
**Pull request by jlewitt1**
Screen Shot 2024-11-18 at 23 44 12

### CODE REVIEW

**Comment by jlewitt1** [comment]
* **#1447** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1447?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1447?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: runhouse/resources/hardware/launcher_utils.py [hunk 1]
```diff
@@ -135,6 +135,10 @@ def run_verbose(
                 spinner.stop()
                 spinner = None
 
+            if event.event == "log_spinner":
+                spinner = Spinner(logger=logger, desc=str(event.data))
```
Comment (mkandler) [inline runhouse/resources/hardware/launcher_utils.py:139]: Would you want to include a line above it to persist in the logs after the spinner stops (similar to `info_spinner`):
```
cls.log_processor.log_event(event, cluster_name=cluster_name)
```

**Review by mkandler — APPROVED** [review]
Just that nit

File: runhouse/resources/hardware/launcher_utils.py (hunk 1 above)
Comment (jlewitt1) [reply runhouse/resources/hardware/launcher_utils.py:139]: actually not sure if we do, bc the log itself is meaningless in that situation (for event type of log spinner i mean). or am i missing something?

Comment (mkandler) [reply runhouse/resources/hardware/launcher_utils.py:139]: Nvm, I think it makes sense without that based on the other PRs!

**Pull request merged**